lib/CleanerPlugin.js

```diff
@@ -0,0 +1,43 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+"use strict";
+const del = require("del");
+
+class CleanerPlugin {
+	constructor(options) {
+		this.options = options || {};
+
+		this.patterns = [];
+		this.cleanerOptions = {};
+
+		if (typeof this.options === "string") {
+			this.patterns = [this.options];
+		} else if (Array.isArray(this.options)) {
+			this.patterns = this.options;
+		} else {
+			this.patterns = this.options.patterns;
+			this.cleanerOptions = this.options.options;
+		}
+	}
+
+	apply(compiler) {
+		compiler.hooks.beforeRun.tapAsync(
```

Comment:
You could also use `tapPromise` and return `del`'s promise instead

Reply:
@ooflorent done :+1: